fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctByCTERule.java

```diff
@@ -124,6 +124,10 @@ public boolean check(OptExpression input, OptimizerContext context) {
             return false;
         }
 
+        if (agg.getLimit() >= 0) {
```

Comment:
don't check none-groupby + limit?

Reply:
in the case of non-groupby, non-cte is faster than cte according to the test results.